website/content/labs/lab6.md

```diff
@@ -0,0 +1,176 @@
+
+`script` specifies the list of commands that will be run as part of this job. Think of this as the commands you would type into the shell if you had to build this program. Note that, in this case, we’re only running one command, but we could specify multiple commands to run in sequence like this:
+  
+    script:
+       - foo
+       - bar
+       - baz
+  
+Take into account that the commands are run from the root of your repository, so we use the `-C lab6/hello/` option to tell `make` to run the Makefile in that directory. We also specified that `make` should run `clean` followed by `all`; this is considered good practice in a build job, as it guarantees that your build starts from a clean slate (and can help you detect whether you’ve inadvertently committed object files or binary files that make your build work only when those files are already present)
+
+`only` specifies the name of the branches where new commits will trigger the running fo the job. In this case, we only want this job to be run if we push anything new to `master`. That means we could create another branch where we allow ourselves to push potentially broken code, and only run the CI jobs when that branch is merged to `master`.
+
+There are actually many other options you can set in `.gitlab-ci.yml`. We will only cover a few others in this lab, but you can see the complete reference here: https://docs.gitlab.com/ee/ci/yaml/
+
+Now, do the following:
+
+* [0 points] Commit the `.gitlab-ci.yml` to your repository and push it.
+
+* [5 points] Go to your repository on the GitLab server (i.e., https://mit.cs.uchicago.edu/cmsc22000-spr-18/CNETID, where CNETID should be replaced by your CNetID). On the left sidebar, click on “Pipelines”. This will show you the pipelines that have been run in your repository. You should see a “passed” pipeline (if you do not, please ask for help). Click on either “passed” or the pipeline number. This will take you to a page with more details about the pipeline. Copy the URL of that page and paste it in the `tasks.txt` file in `lab6/`. This page shows more details about the jobs that ran in that pipeline. In this case, we have only one job: `hello:build`. If you click on it, you’ll be able to see the terminal output from the job (which was run on a special server known as the build server)
+
+* [5 points] Make a change to the `hello.c` file that will make it not build. Commit and push this change to the repository. This will result in a failed pipeline in the “Pipelines” page; copy-paste the URL of the failed pipeline into the `tasks.txt` file. Before moving on to the next task, undo the change and verify that pushing the code results in a “passed” pipeline.
```

Comment:
See above. It would be valuable here to have them identify in their own words where the failure shows up in the pipeline output.